include/crm/common/ipc_attrd.h

```diff
@@ -51,13 +51,17 @@ typedef struct {
  * \brief Send a request to pacemaker-attrd to clear resource failure
  *
  * \param[in] api           Connection to pacemaker-attrd
- * \param[in] host          Affect only this host (or NULL for all hosts)
+ * \param[in] host          Affect only this host
  * \param[in] resource      Name of resource to clear (or NULL for all)
  * \param[in] operation     Name of operation to clear (or NULL for all)
  * \param[in] interval_spec If operation is not NULL, its interval
  * \param[in] user_name     ACL user to pass to pacemaker-attrd
  * \param[in] options       Bitmask of pcmk__node_attr_opts
  *
+ * \note \p host can be NULL to clear failures on all nodes, some host
+ *       name to clear failures only on the given node, or "localhost"
+ *       for the current node.
+ *
```

Comment:
Unfortunately I don't think this approach will work as intended.

The server does accept "localhost" for queries, since NULL means all nodes in that case, but for other request types, NULL should get the local node name while "localhost" will use that literally.

The pcmk__node_attr_target() substitution only works when called from a resource agent, which will have the necessary environment variables set. If called from the command line, "localhost" will be passed as-is to the server with your approach here.

I could be missing something, though. It's hard to keep all the pieces straight.

Reply:
Okay, I've been overthinking this I am pretty sure.  Adding a new commit on top that is much simpler.

Reply:
But, please double check that I'm okay with the change I'm making to crm_attribute for this.